### PR TITLE
new slash mechanism

### PR DESCRIPTION
**Pull request by NathanBSC**
Description
    introduce a new type of slashing for malicious vote.
    it's trigged by crossschain package from systemscontracts in bsc and reuse params of double sign slashing, with following features:
            a. one validator can't be slashed when it has been slashed for malicious vote previous and still in jail
            b. validator can't escape from slashing by editing vote address after malicious vote

Rationale
    

Example
tested in new qa

Changes
NA

### CODE REVIEW

File: x/slashing/params.go [hunk 1]
```diff
@@ -36,18 +39,21 @@ func ParamTypeTable() params.TypeTable {
...
-	SubmitterReward          int64         `json:"submitter_reward"`
-	DowntimeSlashFee         int64         `json:"downtime_slash_fee"`
+	MaxEvidenceAge              time.Duration `json:"max_evidence_age"`
+	SignedBlocksWindow          int64         `json:"signed_blocks_window"`
+	MinSignedPerWindow          sdk.Dec       `json:"min_signed_per_window"`
+	DoubleSignUnbondDuration    time.Duration `json:"double_sign_unbond_duration"`
+	DowntimeUnbondDuration      time.Duration `json:"downtime_unbond_duration"`
+	MaliciousVoteUnbondDuration time.Duration `json:"malicious_vote_unbond_duration"`
+	TooLowDelUnbondDuration     time.Duration `json:"too_low_del_unbond_duration"`
+	SlashFractionDoubleSign     sdk.Dec       `json:"slash_fraction_double_sign"`
```
Comment (NathanBSC) [inline x/slashing/params.go:49]: should we need SlashFractionMaliciousVote to cap penalty？

File: x/slashing/types.go [hunk 2]
```diff
@@ -8,3 +8,10 @@ type SideDowntimeSlashPackage struct {
 	SideChainId   types.ChainID `json:"side_chain_id"`
 	SideTimestamp uint64        `json:"side_timestamp"`
 }
+
+type SideMalicousVoteSlashPackage struct {
+	SideVoteAddr  []byte        `json:"side_vote_addr"`
```
Comment (guagualvcha) [inline x/slashing/types.go:13]: will `[32]byte` better than `[]byte`?

File: x/slashing/keeper.go [hunk 3]
```diff
@@ -376,6 +408,100 @@ func (k *Keeper) slashingSideDowntime(ctx sdk.Context, pack *SideDowntimeSlashPa
...
+		return ErrDuplicateMalicousVoteClaim(k.Codespace)
+	}
+
+	slashAmt := k.MaliciousVoteSlashAmount(sideCtx)
+	validator, slashedAmt, err := k.validatorSet.SlashSideChain(ctx, sideChainName, sideConsAddr, sdk.NewDec(slashAmt))
+	if err != nil {
+		return ErrFailedToSlash(k.Codespace, err.Error())
+	}
+
+	maliciousVoteClaimFee := k.MaliciousVoteSlashFee(sideCtx)
```
Comment (guagualvcha) [inline x/slashing/keeper.go:442]: I think the fee is not needed here, we can simplify it.

File: x/slashing/params.go (hunk 1 above)
Comment (guagualvcha) [reply x/slashing/params.go:49]: The malicious vote is a kind of double sign, may be we can reuse most of the logic of double sign and also the parameters here. Just simplify the feature.

File: x/slashing/keeper.go (hunk 3 above)
Comment (NathanBSC) [reply x/slashing/keeper.go:442]: good idea

File: x/slashing/types.go (hunk 2 above)
Comment (NathanBSC) [reply x/slashing/types.go:13]: in the sdk, use []byte to express SideVoteAddr everywhere
for consistency, I also use  []byte, but [48]byte is better indeed.

File: x/slashing/keeper.go [hunk 4]
```diff
@@ -5,6 +5,7 @@ import (
 	"math"
 	"time"
 
+	"github.com/docker/docker/daemon/logger"
```
Comment (guagualvcha) [inline x/slashing/keeper.go:8]: Please don't import the logger from `docker`

File: x/slashing/keeper.go [hunk 5]
```diff
@@ -237,12 +238,21 @@ func (k *Keeper) SubscribeParamChange(hub types.ParamChangePublisher) {
...
-	}
-	if err != nil {
-		resCode = uint32(err.ABCICode())
+		err = k.slashingSideDowntime(ctx, downTimePack)
+		if err != nil {
+			resCode = uint32(err.ABCICode())
+		}
+	} else {
+		maliciousVotePack, err := k.checkMalicousVoteSlashPackage(payload)
+		logger.Warn("SideMalicousVoteSlashPackage:%v", maliciousVotePack)
```
Comment (guagualvcha) [inline x/slashing/keeper.go:249]: typo here `SideMalicousVoteSlashPackage`

File: x/slashing/types.go (hunk 2 above)
Comment (guagualvcha) [inline x/slashing/types.go:13]: The definition is exactly same with `SideDowntimeSlashPackage`, that is why it wont work.

**Review by guagualvcha — APPROVED** [review]

**Comment by buddh0** [comment]
need squash when merge

**Review by Pythonberg1997 — APPROVED** [review]

**Pull request merged**